docs/proposals/20230407-optional-cluster-infra-resource.md

```diff
@@ -0,0 +1,207 @@
+    - [Risks and Mitigations](#risks-and-mitigations)
+  - [Alternatives](#alternatives)
+  - [Upgrade Strategy](#upgrade-strategy)
+  - [Additional Details](#additional-details)
+    - [Test Plan [optional]](#test-plan-optional)
+    - [Graduation Criteria [optional]](#graduation-criteria-optional)
+    - [Version Skew Strategy [optional]](#version-skew-strategy-optional)
+  - [Implementation History](#implementation-history)
+
+## Glossary
+
+Refer to the [Cluster API Book Glossary](https://cluster-api.sigs.k8s.io/reference/glossary.html).
+
+The following terms will be used in this document.
+
+- `<Infra>Cluster`
+  - When we say `<Infra>Cluster` we refer to any provider's infra-specific implementation of the Cluster API `Cluster` resource spec. When you see `<Infra>`, interpret that as a placeholder for any provider implementation. Some concrete examples of provider infra cluster implementations are Azure's CAPZ provider (e.g., `AzureCluster` and `AzureManagedCluster`), AWS's CAPA provider (e.g., `AWSCluster` and `AWSManagedCluster`), and Google Cloud's CAPG provider (e.g., `GCPCluster` and `GCPManagedCluster`). Rather than referencing any one of the preceding actual implementations of infra cluster resources, we prefer to generalize to `<Infra>Cluster` so that we don't suggest any provider-specific bias informing our conclusions.
+- Managed Kubernetes
+  - Managed Kubernetes refers to any Kubernetes Cluster provisioning and maintenance platform that is exposed by a service API. For example: [EKS](https://aws.amazon.com/eks/), [OKE](https://www.oracle.com/cloud/cloud-native/container-engine-kubernetes/), [AKS](https://azure.microsoft.com/en-us/products/kubernetes-service), [GKE](https://cloud.google.com/kubernetes-engine), [IBM Cloud Kubernetes Service](https://www.ibm.com/cloud/kubernetes-service), [DOKS](https://www.digitalocean.com/products/kubernetes), and many more throughout the Kubernetes Cloud Native ecosystem.
+- _Kubernetes Cluster Infrastructure_
+  - When we refer to _Kubernetes Cluster Infrastructure_ we aim to distinguish required environmental infrastructure (e.g., cloud virtual networks) in which a Kubernetes cluster resides as a "set of child resources" from the Kubernetes cluster resources themselves (e.g., virtual machines that underlie nodes, managed by Cluster API). Sometimes this is referred to as "BYO Infrastructure"; essentially, we are talking about **infrastructure that supports a Kubernetes cluster, but is not actively managed by Cluster API**. As we will see, this boundary is different when discussing Managed Kubernetes: more infrastructure resources are not managed by Cluster API when running Managed Kubernetes.
+- e.g.
+  - This just means "For example:"!
+
+## Summary
+
+We propose to make provider `<Infra>Cluster` resources optional in order to better represent Managed Kubernetes scenarios where all _Kubernetes Cluster Infrastructure_ is managed by the service provider, and not by Cluster API.
```

Comment:
Should this proposal supersede / modify https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/proposals/20220725-managed-kubernetes.md ?

Reply:
That makes the most sense (to me). It would seem beneificial to have the prior document available for folks wishing to do historical analysis of the evolution of managed k8s in CAPI, so maybe we can update that doc in this PR with some explanatory heading that its conclusions are historical only and this new CAEP is the current source of truth?

Reply:
I think the recommendations from the original proposal are still largely the same....ideally that managed kubernetes services should still try to adhere to the split between infra cluster & control plane responsibilities (i.e. option 3) if they have additional infra over the managed k8s service (i.e. like VPCs, Resource Groups). 

The changes in this proposal:

- Make the implementation of option 3 from the original proposal easier.  This is because the control plane endpoint can be reported back via the Control Plane (and not infra cluster) and so the "controller dance" isn't needed
- For an "infrastructure less" (or "Vanilla managed Kubernetes" as @fabriziopandini calls it in the proposal google doc) managed kubernetes service it doesn't require you to specify an infra cluster 
- Support the current implementations in CAPA/CAPZ where the control plane does infra and manage k8s.

So if we are to supersede the original proposal, then we need to cover all the other aspects from the original proposal around the split of responsibilities...imho.

Reply:
Maybe there's a way to update https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/proposals/20220725-managed-kubernetes.md with our recent conclusions and have the current/new proposal to propose the concrete changes needed for core CAPI to support more of the cases?

Reply:
Everyone in this thread is correct. :)

Let's remove the work supersede because that suggests it's replacing it, and that it's "better"... I think this proposal carries on from the prior one and suggests a specific CAPI change that makes the prior proposal (option 3) more practical/flexible.

And as @sbueringer suggests, if we want this CAEP to have the complete, up-to-date recommendation data for Managed Kubernetes going forward, then this CAEP will become huge and I don't think we want that. I was suggesting a sort of composition, where readers are meant to read 1) original proposal, then 2) this CAEP in that order, and synthesize that all as a single source of documentation. That's too hard, let's not do that.

So my view at this point is that we want to make this particular CAEP as targeted as possible to the particular changes to CAPI's Cluster resource reconciliation behavior. Can we then do some appropriate edits to the original proposal to incorporate this new flexibility of not having an infra Cluster resource, or if it's weird to edit a historical document, maybe we can fork a new, updated copy of it with those edits?

Reply:
> or if it's weird to edit a historical document, maybe we can fork a new, updated copy of it with those edits?

Based on precedent, it's fine to edit the document. In the past we tried to keep the proposal up-to-date according to our current implementation (e.g. KCP / ClusterClass proposal). I think that's more useful than a point-in-time outdated state. We always have git history for archaeology.

Reply:
I've started the process of adding some updates to the doc, trying to be as light touch as possible so the original proposal is still readable as a historical document